includes/Core/Notifications/Notifications.php

```diff
@@ -0,0 +1,262 @@
+
+							if ( is_wp_error( $response ) ) {
+								return $response;
+							}
+
+							try {
+								$response = $this->parse_response( $response );
+							} catch ( Exception $e ) {
+								return new WP_Error( 'exception', $e->getMessage() );
+							}
+
+							$data = array_map(
+								function ( Notification $notification ) {
+									return $notification->prepare_for_js();
+								},
+								$this->map_response_to_notifications( $response )
+							);
+
+							return new WP_REST_Response( $data );
+						},
+						'permission_callback' => $can_setup,
+					),
+				)
+			),
+			new REST_Route(
+				'core/site/data/mark-notification',
+				array(
+					array(
+						'methods'             => WP_REST_Server::EDITABLE,
+						'callback'            => function ( WP_REST_Request $request ) {
+							$data = $request['data'];
+
+							if ( empty( $data['notificationID'] ) ) {
+								return $this->missing_required_param( 'data.notificationID' );
+							}
+							if ( empty( $data['notificationState'] ) ) {
+								return $this->missing_required_param( 'data.notificationState' );
+							}
+
+							$credentials = $this->credentials->get();
+							$response    = wp_remote_post(
+								$this->google_proxy->url( '/notifications/mark' ),
+								array(
+									'body' => array(
+										'site_id'         => $credentials['oauth2_client_id'],
+										'site_secret'     => $credentials['oauth2_client_secret'],
+										'notification_id' => $data['notificationID'],
+										'notification_state' => $data['notificationState'],
+									),
+								)
+							);
+
+							if ( is_wp_error( $response ) ) {
+								return $response;
+							}
+
+							try {
+								$response = $this->parse_response( $response );
+							} catch ( Exception $e ) {
+								return new WP_Error( 'exception', $e->getMessage() );
+							}
+
+							return new WP_REST_Response(
+								array(
+									'success' => isset( $response['success'] ) ? (bool) $response['success'] : false,
+								)
+							);
+						},
+						'args'                => array(
+							'data' => array(
+								'required' => true,
+								'type'     => 'object',
+							),
+						),
+						'permission_callback' => $can_setup,
+					),
+				)
+			),
+		);
+	}
+
+	/**
+	 * Validates and parses the given JSON response into an array.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $response HTTP response array.
+	 * @return mixed JSON decoded response.
+	 * @throws Exception Throws exception if response cannot be parsed or if an error is returned.
+	 */
+	private function parse_response( $response ) {
+		$body    = wp_remote_retrieve_body( $response );
+		$decoded = json_decode( $body, true );
+
+		if ( json_last_error() ) {
+			throw new Exception( 'Error while decoding response: ' . json_last_error() );
+		}
+
+		if ( ! empty( $decoded['error'] ) ) {
+			throw new Exception( $decoded['error'] );
+		}
```

Comment:
@felixarntz Should we ever display an error message returned by the remote? Currently it's logged in the console only.

Reply:
A generic error message (like a 500, if the proxy's notification endpoint was down/erroring) seems wrong to display, because it's not very actionable/helpful to the user. I'd think an allow-list of errors to display would work if wanted, but otherwise it seems best to log them to the console for debugging/support. Otherwise hide 'em?

Reply:
Yeah I think we should stick to only logging and don't render any user-facing messages. It should just silently error then.